Makefile

```diff
@@ -76,6 +76,7 @@ genesis_ledger:
 	$(info Genesis ledger and genesis proof generated)
 
 build: git_hooks reformat-diff libp2p_helper
+	if ! ocaml --version | grep 4.11.2; then echo "incorrect ocaml version"; exit 1; fi
```

Comment:
Two suggestions: 
- use a variable for the target OCaml version, rather than inlining it
- make the version check its own target, so it can be added as a prerequisite for other commands